src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,114 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isAdding: boolean;
+  onDeleteItem: (id: number) => void;
+  onToggleTodo: (id: number, completed: boolean) => void;
+  onRenameTodo: (todo: Todo, newTitle: string) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isAdding,
+  onDeleteItem,
+  onToggleTodo,
+  onRenameTodo,
+}) => {
+  const { completed, id, title } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const currentTitle = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (currentTitle.current) {
+      currentTitle.current.focus();
+    }
+  });
+
+  const onSuccessfulEdit = useCallback(() => {
+    onRenameTodo(todo, newTitle);
+    setIsEditing(false);
+  }, [newTitle]);
+
+  const onPressEsc = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement>) => {
+      if (event.key === 'Escape') {
+        setIsEditing(false);
+        setNewTitle(todo.title);
```

Comment:
You have destructured value from todo in 25 line. Why do you use `todo.`? Fix all cases please